crates/rslint_core/src/groups/errors/no_confusing_arrow.rs

```diff
@@ -0,0 +1,115 @@
+use crate::rule_prelude::*;
+use ast::{ArrowExpr, Expr};
+
+declare_lint! {
+    /**
+    Disallow arrow functions where they could be confused with comparisons.
+
+    Arrow functions (`=>`) are similar in syntax to some comparison operators (`>`, `<`, `<=`, and `>=`).
+    This rule warns against using the arrow function syntax in places where it could be confused with
+    a comparison operator
+
+    Here's an example where the usage of `=>` could be confusing:
+
+    ```js
+    // The intent is not clear
+    var x = a => 1 ? 2 : 3;
+    // Did the author mean this
+    var x = function (a) { return 1 ? 2 : 3 };
+    // Or this
+    var x = a >= 1 ? 2 : 3;
+    ```
+
+    ## Incorrect Code Examples
+
+    ```js
+    var x = a => 1 ? 2 : 3;
+    var x = (a) => 1 ? 2 : 3;
+    ```
+    */
+    #[serde(default)]
+    NoConfusingArrow,
+    errors,
+    "no-confusing-arrow",
+    /// Relaxes the rule and accepts parenthesis as a valid "confusion-preventing" syntax.
+    pub allow_parens: bool
+}
+
+impl Default for NoConfusingArrow {
+    fn default() -> Self {
+        Self { allow_parens: true }
+    }
+}
+
+#[typetag::serde]
+impl CstRule for NoConfusingArrow {
+    fn check_node(&self, node: &SyntaxNode, ctx: &mut RuleCtx) -> Option<()> {
+        let function_stmt = node.try_to::<ArrowExpr>()?;
+        let expr = Expr::cast(function_stmt.body()?.syntax().to_owned())?;
```

Comment:
```suggestion
        let expr = function_stmt.child_with_ast::<Expr>()?;
```
This can be shortened to this since expr wont appear anywhere else in the top level of the arrow expr node

Reply:
Sounds good 👍

Reply:
I try to implement with your suggestion, I found in some cases it fails.

```
a => b ? 2 : 3;
^
```

This is a `NAME_REF` which is one of `Expr`, so `child_with_ast` returns the `NAME_REF` node.
but in this case, I want to grab the behind of `ARROW_EXPR`. 

```
ARROW_EXPR@0..14
  NAME_REF@0..1        <- child_with_ast returns this
    IDENT@0..1 "a"

  FAT_ARROW@2..4 "=>"

  COND_EXPR@5..14      <- but I want this
  ...
```

I think I can use the `try_to` instead of the `Expr::cast`. How about this?

```suggestion
        let expr = Expr::cast(function_stmt.body()?.syntax().to_owned())?;
        let expr = function_stmt.body()?.syntax().try_to::<Expr>()?;
```

Reply:
Oh hmmm, that should not be `NAME_REF`, that should be `NAME` because its binding 🤔

Reply:
Yeah thats a bug, im fixing it right now 🙂

Reply:
> Yeah thats a bug, im fixing it right now 🙂

OK, thanks! Now I change this line using `try_to` instead of `Expr::cast`.

Reply:
Alright i fixed it in the latest commit so if you rebase then that method should work

Reply:
Oh, great...